core/src/main/java/org/jdbi/v3/core/statement/ColonPrefixSqlParser.java

```diff
@@ -36,25 +36,24 @@
  * This is the default SQL parser
  * </p>
  */
-public class ColonPrefixSqlParser implements SqlParser {
-    private static final JdbiCache<String, ParsedSql> PARSED_SQL_CACHE =
-        JdbiCaches.declare(ColonPrefixSqlParser::internalParse);
+public class ColonPrefixSqlParser extends CachingSqlParser {
 
-    @Override
-    public ParsedSql parse(String sql, StatementContext ctx) {
-        try {
-            return PARSED_SQL_CACHE.get(sql, ctx);
-        } catch (IllegalArgumentException e) {
-            throw new UnableToCreateStatementException("Exception parsing for named parameter replacement", e, ctx);
-        }
+    public ColonPrefixSqlParser() {
+        this(Caffeine.newBuilder());
+    }
+
+    @Beta
+    public ColonPrefixSqlParser(Caffeine<Object, Object> cache) {
```

Comment:
This adds Caffeine to our public API. I'm ok with this as `@Beta` but we should be very confident about this decision before making it permanent (i.e. removing the Beta annotation).

Reply:
Yup, agreed.  I wasn't sure of a good way to expose Caffeine configuration otherwise, and I did not want to create a "Jdbi cache config" interface.

Reply:
Perhaps all you need is a loading function, e.g. `Function<K, V>`, as settable in the public API. Then `cache::get` can be passed in to delegate to caffeine, for instance. This of course reduces the flexibility of how you can use the cache (e.g. no `asMap` view) but decouples with minimal work.

Reply:
I like the idea of putting a facade API in front of the cache provider. Then somebody could configure e.g. a provider that performs no caching if their hit rate is too low.

Reply:
Similar to how we have TemplateEngine as an abstraction in front of FreeMarker and StringTemplate.